_pytest/python.py

```diff
@@ -248,8 +248,10 @@ def pytest_pycollect_makeitem(collector, name, obj):
         if collector.classnamefilter(name):
             Class = collector._getcustomclass("Class")
             outcome.force_result(Class(name, parent=collector))
-    elif collector.funcnamefilter(name) and hasattr(obj, "__call__") and\
-        getfixturemarker(obj) is None:
+    elif (
```

Comment:
That condition is starting to be waaay to complex, can you turn it into a method?